DependencyInjection/Configuration.php

```diff
@@ -195,6 +195,9 @@ private function getDbalConnectionsNode(): ArrayNodeDefinition
                     ->useAttributeAsKey('name')
                     ->prototype('scalar')->end()
                 ->end()
+                ->scalarNode('schema_manager_factory')
+                    ->defaultValue('doctrine.dbal.default_schema_manager_factory')
```

Comment:
@derrabus do you think its a good idea to use `Doctrine\DBAL\Schema\DefaultSchemaManagerFactory` by default?

Or should we make this opt-in?

Reply:
This will break apps that:

* Use a custom DBAL driver.
* Override the schema manager via a custom platform class.

I'll let you decide whether this is acceptable. 🙂

Reply:
Given that the DefaultSchemaManagerFactory is a BC-breaking solution, it should not be the default in DoctrineBundle for now (unless DBAL 4.x is used).

I suggest having a simple boolean flag to opt-in for the new default, as done in several places in Symfony for such migrations (see how SecurityBundle 5.3 introduced  a boolean flag to opt-in for the new security system).

The behavior of that flag should be like that:

- when using DBAL 3.6+, trigger a deprecation warning if the flag is not set or is set to using the legacy behavior
- when using DBAL 4+, throw an exception if the flag is set to `false`

and then, make a PR on the symfony/recipes repo configuring the flag to use the new behavior, so that new projects start with the new behavior.

Reply:
I don't see why it should be a boolean. We can make it use legacy factory by default if it exist, otherwise default factory.

Reply:
@ostrolucky yeah I was also thinking to keep the setting but using legacy as default + set the new default factory then via recipe

Reply:
well, if we want to support custom factories instead of only the legacy one and the new one, this can indeed be done with a non-boolean config

Reply:
The possibility of custom factories is kinda the whole point of this new setting: it enables the app to override the schema manager.